src/codemagic/tools/firebase_app_distribution/argument_types.py

```diff
@@ -0,0 +1,23 @@
+import argparse
+import json
+from typing import Dict
+
+from codemagic import cli
+
+
+class CredentialsArgument(cli.EnvironmentArgumentValue[dict]):
+    argument_type = dict
+    environment_variable_key = 'FIREBASE_SERVICE_ACCOUNT_CREDENTIALS'
+
+    @classmethod
+    def _apply_type(cls, non_typed_value: str) -> Dict:
+        try:
+            return json.loads(non_typed_value)
```

Comment:
See `_apply_type` definition in parent. It does all the validations and hence it shouldn't return that early.

Reply:
This still holds.